packages/@react-aria/utils/src/useViewportSize.ts

```diff
@@ -88,10 +88,19 @@ export function useViewportSize(): ViewportSize {
   return size;
 }
 
+/**
+ * Get the viewport size without the scrollbar.
+ */
 function getViewportSize(): ViewportSize {
   return {
     // Multiply by the visualViewport scale to get the "natural" size, unaffected by pinch zooming.
-    width: visualViewport ? visualViewport.width * visualViewport.scale : window.innerWidth,
-    height: visualViewport ? visualViewport.height * visualViewport.scale : window.innerHeight
+    width: visualViewport
+      // The visual viewport width may include the scrollbar gutter width.
+      // See: https://github.com/w3c/csswg-drafts/issues/8099
+      ? Math.min(visualViewport.width * visualViewport.scale, document.documentElement.clientWidth)
+      : document.documentElement.clientWidth,
```

Comment:
The visual viewport size does not include the scrollbar. Changed it from `window.innerWidth` to `document.documentElement.clientWidth` to get the same behavior.

See: https://drafts.csswg.org/cssom-view/#visualViewport

> Otherwise, return the width of the [visual viewport](https://drafts.csswg.org/cssom-view/#visual-viewport) excluding the width of any rendered vertical [classic scrollbar](https://drafts.csswg.org/css-overflow-3/#classic-scrollbars) that is fixed to the visual viewport.

Reply:
Split into https://github.com/adobe/react-spectrum/pull/9384